packages/react/src/products/hooks/useProductsList.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  fetchListing,
+  fetchSet,
+  getProductsListError,
+  getProductsListProducts,
+  getProductsListResult,
+  isProductsListLoading,
+} from '@farfetch/blackout-redux/products';
+import { generateProductsListHash } from '@farfetch/blackout-redux/products/utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import type {
+  ListingQuery,
+  SetQuery,
+} from '@farfetch/blackout-client/products/types';
+import type {
+  ProductsListTypes,
+  UseProductsList,
+  UseProductsListParams,
+} from './types/useProductsList';
+
+export const PRODUCTS_LIST_TYPES: ProductsListTypes = {
+  LISTING: 'listing',
+  SET: 'set',
+};
+
+/**
+ * Hook to handle listing logic.
+ *
+ * @param {object} props - Object containing the necessary info to use inside the hook.
+ * @param {string} [props.slug] - The slug of the given product list.
+ * @param {Object} [props.query] - The query parameters of the given product list.
+ * @param {string} [props.type='listing'] - Listing type ('listing' or 'set').
+ * @param {boolean} [props.useCache=true] - To use cache when fetching a product list.
+ * @param {boolean} [props.setProductsListHash=true] - Wether to set the reducer's hash on the store when requesting a product list.
+ *
+ * @returns {Object} All the info needed to get listing results and information.
+ */
+const useProductsList = ({
+  query,
+  setProductsListHash = true,
+  slug,
+  type = PRODUCTS_LIST_TYPES.LISTING,
+  useCache = true,
+}: UseProductsListParams): UseProductsList => {
+  const isSetPage = type === PRODUCTS_LIST_TYPES.SET;
+  const dispatch = useDispatch();
+  const productsListHash = generateProductsListHash(slug, query, {
+    isSet: isSetPage,
+  });
+  const error = useSelector(state =>
+    getProductsListError(state, productsListHash),
+  );
+  const isLoading = useSelector(state =>
+    isProductsListLoading(state, productsListHash),
+  );
+  const result = useSelector(state =>
+    getProductsListResult(state, productsListHash),
+  );
+  const products = useSelector(state =>
+    getProductsListProducts(state, productsListHash),
+  );
+
+  useEffect(() => {
+    if (slug !== undefined && !products) {
+      dispatch(
+        isSetPage
+          ? fetchSet(slug, query as SetQuery, { useCache, setProductsListHash })
+          : fetchListing(slug, query as ListingQuery, {
+              useCache,
+              setProductsListHash,
+            }),
+      );
+    }
+  }, [
+    dispatch,
+    isSetPage,
+    products,
+    query,
+    setProductsListHash,
+    slug,
+    useCache,
+  ]);
+
+  return {
+    error,
+    isLoading,
+    productsListHash,
+    products,
+    result,
```

Comment:
And jsdoc here, or did we decide on not using it anymore?

Reply:
Added! 😄